libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowManager.m

```diff
@@ -32,11 +32,81 @@
 #import "SFApplicationHelper.h"
 #import "SFSecurityLockout.h"
 
+@interface SFSDKUIWindow ()
+- (instancetype)initWithFrame:(CGRect)frame;
+- (instancetype)initWithFrame:(CGRect)frame andName:(NSString *)windowName;
+- (void)stashRootViewController;
+- (void)unstashRootViewController;
+
+@property (nonatomic,strong) UIViewController * _Nullable stashedController;
+@property (nonatomic,readonly) NSString * _Nullable windowName;
+@end
+
+@implementation SFSDKUIWindow
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
+        _windowName = @"NONAME";
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame andName:(NSString *)windowName {
+    self = [super initWithFrame:frame];
+    if (self) {
+        _windowName = windowName;
+    }
+    return self;
+}
+
+- (void)stashRootViewController {
+    if (self.rootViewController) {
+        _stashedController = self.rootViewController;
+        super.rootViewController = nil;
+    }
+}
+
+- (void)unstashRootViewController {
+    if (_stashedController)
+        super.rootViewController = _stashedController;
+}
+
+- (void)setRootViewController:(UIViewController *)rootViewController {
+    _stashedController = rootViewController;
+    super.rootViewController = rootViewController;
+}
+
+- (void)makeKeyAndVisible {
+    [super makeKeyAndVisible];
+}
+
+- (void)becomeKeyWindow {
+    [self unstashRootViewController];
+    if (self.windowLevel <0)
+        self.windowLevel = self.windowLevel * -1;
+    self.alpha = 1.0;
+}
+
+- (void)resignKeyWindow{
+    if ([SFApplicationHelper sharedApplication].applicationState == UIApplicationStateActive){
+        if (self.windowLevel>0)
+            self.windowLevel = self.windowLevel * -1;
+        self.alpha = 0.0;
+        super.rootViewController = nil;
+        [self stashRootViewController];
+    }
+}
+@end
+
 @interface SFSDKWindowManager()<SFSDKWindowContainerDelegate>
 
 @property (nonatomic, strong) NSHashTable *delegates;
 @property (nonatomic, strong,readonly) NSMapTable<NSString *,SFSDKWindowContainer *> * _Nonnull namedWindows;
-@property (nonatomic, strong,readonly) NSMapTable<UIWindow *,SFSDKWindowContainer *> * _Nonnull reverseLookupTable;
+@property (nonatomic,weak) SFSDKWindowContainer *lastActiveWindow;
```

Comment:
Moved snapshot handling logic to the WindowManager. Capture the last active window when the snapshot is being present and restore when its dismissed.